File: python-code/ex2/carro.py
class Carro:
  def __init__(self, marca, modelo, ano):
    self.marca = marca
    self.modelo = modelo
    self.ano = ano
    self.velocidade = 0 

  def acelerar(self, incremento):  
    self.velocidade += incremento
    print(f"O carro acelerou. Velocidade atual: {self.velocidade} km/h")

  def frear(self, decremento):
    self.velocidade -= decremento

    if self.velocidade < 0:
      self.velocidade = 0
      
    print(f"O carro freou. Velocidade atual: {self.velocidade} km/h")

  def exibir_velocidade(self):
    print(f"Velocidade atual: {self.velocidade} km/h")


carro = Carro("BMW", "M4", 2022)

carro.acelerar(60)   
carro.acelerar(150)
carro.frear(20)  
carro.exibir_velocidade() 
carro.frear(30)
carro.acelerar(50)
carro.exibir_velocidade()

File: java-code/src/main/java/ex2/Carro.java
package ex2;

public class Carro {
    String marca;
    String modelo;
    int ano;
    int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    void acelerar(int incremento) {
        velocidade += incremento;
        System.out.println("O carro acelerou. Velocidade atual: " + velocidade + " km/h");
    }

    void frear(int decremento) {
        velocidade -= decremento;
        if (velocidade < 0) {
            velocidade = 0;
        }
        System.out.println("O carro freou. Velocidade atual: " + velocidade + " km/h");
    }

    void exibirVelocidade() {
        System.out.println("Velocidade atual: " + velocidade + " km/h");
    }
}
